### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by galactoise**

### CODE REVIEW

File: service/src/main/java/com/derpgroup/quip/manager/QuipManager.java [hunk 1]
```diff
@@ -25,6 +26,8 @@
     ConversationHistoryUtils.getMapper().registerModule(new MixInModule());
   }
   
+  Logger logger = LoggerFactory.getLogger(QuipManager.class);
```
Comment (davidwphillips) [inline service/src/main/java/com/derpgroup/quip/manager/QuipManager.java]: Such lowercase, much shame, very ruffled. Wow

Comment (galactoise) [inline service/src/main/java/com/derpgroup/quip/manager/QuipManager.java]: Actually, it looks like we have both LOG and logger in this branch currently.

File: service/src/main/java/com/derpgroup/quip/bots/complibot/resource/CompliBotAlexaResource.java [hunk 2]
```diff
@@ -81,60 +87,67 @@ public CompliBotAlexaResource(MainConfig config, Environment env) {
...
-      VoiceOutput<SsmlOutputSpeech> voiceOutput = (VoiceOutput<SsmlOutputSpeech>) VoiceMessageFactory.buildOutputMessage(builder.build(), InterfaceType.ALEXA);
-      outputSpeech = voiceOutput.getImplInstance();
-    }else{
-      outputSpeech = null;
-      card = null;
-      speechletResponse.setShouldEndSession(true);
+    QuipMetadata metadata = null;
+    try {
+      if(testFlag == null || testFlag == false){ 
+        AlexaUtils.validateAlexaRequest(request, signatureCertChainUrl, signature);
```
Comment (davidwphillips) [inline service/src/main/java/com/derpgroup/quip/bots/complibot/resource/CompliBotAlexaResource.java]: This function will blow up if request is null. Our nullity check should be above here?

File: service/src/main/java/com/derpgroup/quip/bots/insultibot/resource/InsultiBotAlexaResource.java [hunk 3]
```diff
@@ -81,60 +87,67 @@ public InsultiBotAlexaResource(MainConfig config, Environment env) {
...
-      VoiceOutput<SsmlOutputSpeech> voiceOutput = (VoiceOutput<SsmlOutputSpeech>) VoiceMessageFactory.buildOutputMessage(builder.build(), InterfaceType.ALEXA);
-      outputSpeech = voiceOutput.getImplInstance();
-    }else{
-      outputSpeech = null;
-      card = null;
-      speechletResponse.setShouldEndSession(true);
+    QuipMetadata metadata = null;
+    try {
+      if(testFlag == null || testFlag == false){ 
+        AlexaUtils.validateAlexaRequest(request, signatureCertChainUrl, signature);
```
Comment (davidwphillips) [inline service/src/main/java/com/derpgroup/quip/bots/insultibot/resource/InsultiBotAlexaResource.java]: Same thing for here with the null check first

**Comment by galactoise** [comment]
@davidwphillips nullchecks fixed

**Pull request merged**